test/lit/passes/gsi-nontype.wast

```diff
@@ -267,3 +267,61 @@
   )
 )
 
+;; Nested struct.gets that seem optimizable.
+(module
+  (rec
+    ;; CHECK:      (rec
+    ;; CHECK-NEXT:  (type $func (func (result anyref)))
+    (type $func (func (result anyref)))
+    ;; CHECK:       (type $outer (sub (struct (field (ref $inner)))))
+    (type $outer (sub (struct (field (ref $inner)))))
+    ;; CHECK:       (type $inner (sub (struct (field (ref $func)))))
+    (type $inner (sub (struct (field (ref $func)))))
+  )
+
+  ;; CHECK:      (type $3 (func (result anyref)))
+
+  ;; CHECK:      (global $global.unnested.0 (ref (exact $inner)) (struct.new $inner
+  ;; CHECK-NEXT:  (ref.func $func)
+  ;; CHECK-NEXT: ))
+
+  ;; CHECK:      (global $global (ref $outer) (struct.new $outer
+  ;; CHECK-NEXT:  (global.get $global.unnested.0)
+  ;; CHECK-NEXT: ))
+  (global $global (ref $outer) (struct.new $outer
+    (struct.new $inner
+      (ref.func $func)
+    )
+  ))
+
+  ;; CHECK:      (func $func (type $func) (result anyref)
+  ;; CHECK-NEXT:  (unreachable)
+  ;; CHECK-NEXT: )
+  (func $func (type $func) (result anyref)
+    (unreachable)
+  )
+
+  ;; CHECK:      (func $caller (type $3) (result anyref)
+  ;; CHECK-NEXT:  (call_ref $func
+  ;; CHECK-NEXT:   (struct.get $inner 0
+  ;; CHECK-NEXT:    (global.get $global.unnested.0)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $caller (result anyref)
+    (call_ref $func
```

Comment:
Is the `call_ref` important for the optimization, or just to emulate a typical usage we're interested in optimizing?